libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -535,12 +550,125 @@ void AP_ADSB::send_dynamic_out(const mavlink_channel_t chan)
             squawk);
 }
 
+uint32_t AP_ADSB::get_encoded_icao(void)
+{
+    // utilize the upper unused 8bits of the icao with special flags.
+    // This encoding is required for uAvionix devices that break the MAVLink spec.
```

Comment:
Isn't the MAVLink spec for these messages dictated by uAvionix? Why isn't the spec changed? Seems quite strange to hide this away in ArduPilot code and not document it in the MAVLink spec.